src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/OnboardingSupervisor.kt

```diff
@@ -621,8 +620,13 @@ internal class OnboardingSupervisor(
 
             val result = helper.parser.decodeJsonObject(response)
             if (result != null) {
+                val usdcDecimals = helper.environment.tokens["usdc"]?.decimals ?: 6
+                val nativeDecimals = helper.environment.tokens["chain"]?.decimals ?: 18
+                val nativeDenom = helper.environment.tokens["chain"]?.denom
                 val amountMap =
                     helper.parser.asMap(helper.parser.asList(result["amount"])?.firstOrNull())
+                val denom = helper.parser.asString(amountMap?.get("denom"))
+                val decimals = if (denom === nativeDenom) nativeDecimals else usdcDecimals
                 val amount = helper.parser.asDecimal(amountMap?.get("amount"))
                 val usdcAmount = amount?.div(Numeric.decimal.TEN.pow(decimals))
```

Comment:
`usdcAmount` here would be populated incorrectly even with the correct decimals.

In a later comparison, f the user is using `adydx` as their denom `usdcAmount`, it should not be subtracted from the `amount` input for validation